notebooks/amazon_forest_notebook.py

```diff
@@ -190,119 +191,121 @@
 
-# <markdowncell>
+input_tensor = Input(shape=img_dim)
 
-# ## Choose Hyperparameters
-# 
-# Choose your hyperparameters below for training. 
-# 
-# Note that we have created a learning rate annealing schedule with a series of learning rates as defined in the array `learn_rates` and corresponding number of epochs for each `epochs_arr`. Feel free to change these values if you like or just use the defaults.
-# 
-# If you opted for a high `img_resize` then you may want to lower the `batch_size` to fit your images matrices into the GPU memory. With an `img_resize` of `(256, 256)` (the default size) and a batch_size of `64` you should at least have a GPU with 8GB or VRAM.
+base_model = VGG16(include_top=False,
+                   weights='imagenet',
+                   input_shape=img_dim)
 
-# <codecell>
+bn = BatchNormalization()(input_tensor)
+x = base_model(bn)
+x = Flatten()(x)
+output = Dense(17, activation='sigmoid')(x)
+
+model = Model(input_tensor, output)
 
-batch_size = 64
-epochs_arr = [35, 15, 5]
-learn_rates = [0.002, 0.0002, 0.00002]
+model.summary()
 
 # <markdowncell>
 
-# ## Define and Train model
-# 
-# Here we define the model and begin training. 
+# ## Fine-tune conv layers
+# Now that we have a trained classifier, we can unfreeze all other layers in the model (or keep certain layers still frozen) and retrain. 
 
 # <codecell>
 
-classifier = AmazonKerasClassifier(preprocessor)
-classifier.add_conv_layer()
-classifier.add_flatten_layer()
-classifier.add_ann_layer(len(preprocessor.y_map))
+from keras.optimizers import Adam
+from keras.callbacks import ModelCheckpoint, EarlyStopping, ReduceLROnPlateau
 
-train_losses, val_losses = [], []
-for learn_rate, epochs in zip(learn_rates, epochs_arr):
-    tmp_train_losses, tmp_val_losses, fbeta_score = classifier.train_model(learn_rate, epochs, batch_size, 
-                                                                           train_callbacks=[checkpoint, fbeta, csv])
-    train_losses += tmp_train_losses
-    val_losses += tmp_val_losses
+callbacks = [EarlyStopping(monitor='val_loss', patience=2, verbose=1, min_delta=1e-4),
+             ReduceLROnPlateau(monitor='val_loss', factor=0.5, patience=1, cooldown=0, verbose=1),
+             ModelCheckpoint(filepath='weights/weights.best.hdf5', verbose=1, save_best_only=True, 
+                             save_weights_only=True, mode='auto')]
 
-print(fbeta.fbeta)
+X_train, y_train = preprocessor.X_train, preprocessor.y_train
+X_val, y_val = preprocessor.X_val, preprocessor.y_val
 
-# <markdowncell>
+epochs_arr = [5, 3, 2]
+learn_rates = [1e-4, 1e-5, 1e-6]
+batch_size = 128
+train_generator = preprocessor.get_train_generator(batch_size)
+steps = len(X_train) // batch_size
 
-# ## Load Best Weights
+for learn_rate, epochs in zip(learn_rates, epochs_arr):
+    model.compile(optimizer=Adam(lr=learn_rate), loss='binary_crossentropy', metrics = ['accuracy'])
+    model.fit_generator(train_generator, steps, epochs, verbose=1, 
```

Comment:
Strange I actually did include the weights folder in the repo when I committed but I guess it didn't show up because the folder was empty like you say. So this time I'll make sure there is a file in there :)